exec/totemconfig.c

```diff
@@ -1925,10 +1934,11 @@ int totem_config_validate (
 	/* Only knet allows crypto */
 	if (totem_config->transport_number != TOTEM_TRANSPORT_KNET) {
 		if ((strcmp(totem_config->crypto_cipher_type, "none") != 0) ||
-		    (strcmp(totem_config->crypto_hash_type, "none") != 0)) {
+		    (strcmp(totem_config->crypto_hash_type, "none") != 0) ||
+		    (totem_config->secauth != 0)) {
```

Comment:
Is `totem_config->secauth` really needed? For example here, when somebody sets both secauth to `on` and hash/crypto to `none` then corosync exists with error eventho it shouldn't. But maybe I'm missing some use case where it is useful and if so, please do hesitate to share your view.

Reply:
hmm. it's a tradeoff between
- explicitly check and not rely on crypto_* != none being implied by secauth
- false positives for secauth == on, but explicit crypto_* == none

I'd rather have the explicit check, but if you are opposed we can drop it in v2..

Reply:
Ok, I see and agree to some extend. Let me add another argument against it. Adding `totem_config->secauth` imho makes source code slightly more confusing (especially because it is in totem_config structure) and harder to test. Without it, there will be only one function which must handle existence of secauth and this function just changes authoritative `crypto_*`.

I know in context of today it looks obvious, but my expiriences tells me, that in a few years there will be another bunch of code and nobody will remember what exactly was relation between `secauth` and `crypto_*` in `totem_config` structure.

So I would rather see drop of  `totem_config->secauth` (if there is no other reason).

Reply:
dropped in v2